src/silo/preprocessing/lineage_definition_file.cpp

```diff
@@ -0,0 +1,129 @@
+#include "silo/preprocessing/lineage_definition_file.h"
+
+#include <fstream>
+
+#include <yaml-cpp/yaml.h>
+
+#include "silo/common/panic.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+
+namespace YAML {
+using silo::common::LineageName;
+using silo::preprocessing::LineageDefinition;
+using silo::preprocessing::LineageDefinitionFile;
+
+template <>
+struct convert<LineageName> {
+   static bool decode(const Node& node, LineageName& lineage_name) {
+      try {
+         lineage_name.string = node.as<std::string>();
+      } catch (const YAML::BadConversion& e) {
+         throw silo::preprocessing::PreprocessingException(fmt::format(
+            "Could not parse Lineage definition name '{}', as it is not a string.", YAML::Dump(node)
+         ));
+      }
+      return true;
+   }
+   static Node encode(const LineageName& lineage_name) {
+      Node node;
+      node = lineage_name.string;
+      return node;
+   }
+};
```

Comment:
Hmm, I just wanted to do this, but it is kind of weird, as the error message is very specific to the location where this is used. As this is not in the header file, I would say that this is just a implementation detail of the lineage_definition_file, therefore having it remain in here. I could put this in a function with a different name (and then also write a custom converter for vector<LineageName>, which calls that function), but putting it in the lineage_name.cpp file would mix concerns IMO

Reply:
Note that this would require us to also put this definition in the header file, "informing" all other files that include lineage_name.cpp what _YAML parsing is_ (including the YAML headers..)

Reply:
Ok, no big preference from my side, let's leave it as it is :+1: